src/interp/interp.cc

```diff
@@ -2084,6 +2084,11 @@ RunResult Thread::DoReinterpret() {
 
 RunResult Thread::DoMemoryInit(Instr instr, Trap::Ptr* out_trap) {
   Memory::Ptr memory{store_, inst_->memories()[instr.imm_u32x2.fst]};
+  TRAP_IF(inst_->datas().size() <= instr.imm_u32x2.snd,
+      StringPrintf("out of bounds data access: access at %u "
+                    ">= max value %" PRIu64,
+                    instr.imm_u32x2.snd,
+                    inst_->datas().size()));
```

Comment:
Shouldn't validation make this impossible?

Reply:
I'm not sure the interpreter should be enforcing constraints that the validator should have already enforced... otherwise we will end up duplicating all the validation rules here.

Reply:
There are some formatting issue in the CI log, so I have chosen to use `%zu` to format vector's size correctly.